cogs/welcome/welcome.py

```diff
@@ -132,6 +200,153 @@ async def logServerLeave(self, leaveUser: discord.Member):
     async def welcome(self, ctx: Context):
         """Server welcome message settings."""
 
+    # [p]greetings
+    @welcome.group(name="greetings")
+    @commands.guild_only()
+    @checks.mod_or_permissions()
+    async def greetings(self, ctx: Context):
+        """Server greetings message settings."""
+
+    @checks.mod_or_permissions()
+    @greetings.command(name="channel")
+    async def welcomeChannelSet(self, ctx: Context, channel: discord.TextChannel = None):
+        """
+        Set the welcome channel
+
+        Parameters:
+        -----------
+        channel: The text channel to set welcome's to. If not passed anything, will remove the welcome channel
+        """
+        if not channel:
+            # channel == None
+            await self.config.guild(ctx.guild).welcomeChannelSet.set(False)
+            await ctx.send("Welcome channel has been removed")
+            return
+
+        await self.config.guild(ctx.guild).welcomeChannel.set(channel.id)
+        await self.config.guild(ctx.guild).welcomeChannelSet.set(True)
+        await ctx.send(f"Channel set to {channel}")
+
+        return
+
+    @checks.mod_or_permissions()
+    @greetings.command(name="add")
+    async def greetAdd(self, ctx: Context, name: str):
+        """
+        Add a new greeting
+
+        Including {USER} in the message will have that replaced with a ping to the new user
+
+        Parameters:
+        -----------
+        name: name of the greeting
+        """
+
+        def check(message: discord.Message):
+            return message.author == ctx.message.author and message.channel == ctx.message.channel
+
+        await ctx.send("What would you like the greeting message to be?")
+        try:
+            greeting = await self.bot.wait_for("message", check=check, timeout=30.0)
+        except asyncio.TimeoutError:
+            await ctx.send("No response received, not setting anything!")
+            return
+
+        # check for if the message with a replaced {USER} will be too long
+        tempmsg = greeting.contents.replace("{USER}", "<@0000000000000000000>")
```

Comment:
Sorry if it wasn't clear, I was actually referring to use `ctx.author.mention` instead, which would be a real one, I don't know how long the ID actually is.
```suggestion
        tempmsg = greeting.contents.replace("{USER}", ctx.author.mention)
```

Reply:
done